README.md

```diff
@@ -14,12 +14,23 @@ If there is anything you are looking for that is not covered here you should ref
 
 ### File structure ###
 
-You should not define multiple public/internal types (ie class, struct, enum) in the same file; each type should have its own file.
+You should not define multiple public/internal types (ie class, struct, enum) in the same file; each type should have its own file. If an enum or protocol is related to the class but has to be known outside define it inside the class file.
+
+E.g., The following enum can be used for both networking and data storage, so it should live as it's own entity:
+
+```swift
+enum Result<T> {
+    case Success(T)
+    case Failure(String)
+}
+```
 
 The following list should be the standard organization of all your Swift files, in this specific order:
 
 Before the type declaration:
 
+* Enum (if only related to the file's class)
+* Protocol (if only related to the file's class)
```

Comment:
Same as above with enums. What defines "related" to one class? Also, why would you only declare a protocol if it's used in one place? Protocols should get their own file too, especially since protocols can have extensions -- where do those go?